test/models/tf_ov_model_zoo/run_infer_single.sh

```diff
@@ -57,83 +77,174 @@ function get_model_repo {
         git checkout ${COMMIT} || exit 1
         gen_frozen_models ./model_factory/create.all
         echo Downloaded all models; echo
-    else
-        cd ${LOCALSTORE} || exit 1
-        git checkout ${COMMIT} || exit 1
-        if [ -d "temp_build" ]; then rm -rf temp_build; fi
-        if [ ! -f "${LOCALSTORE}/frozen/${MODEL}.pb" ] || [ ! -f "${LOCALSTORE}/frozen/${MODEL}.txt" ]; then
-            gen_frozen_models ./model_factory/create_${MODEL}.sh
-            echo Downloaded model ${MODEL}; echo
+    fi
+
+    cd ${LOCALSTORE} || exit 1
+    commit_check=$(git cat-file -t $COMMIT 2>/dev/null)
+    [ "$commit_check" == "commit" ] || ( git fetch; git checkout ${COMMIT} || exit 1; )
+    
+    if [ -d "temp_build" ]; then rm -rf temp_build; fi
+
+    if [ ! -f "${LOCALSTORE}/frozen/${MODEL}.pb" ] || \
+    [ ! -f "${LOCALSTORE}/frozen/${MODEL}.txt" ] || \
+    [ "$(file_newer_than_commit ${LOCALSTORE}/frozen/${MODEL}.pb)" == "0" ]; then
```

Comment:
Wouldn't this always be true? Also, it's checking the timestamp of the file with the timestamp of the latest commit of the repo?

Reply:
This way, if we update the model-zoo commit-id, it will re-create all models in CI agents. This might be needed, as the zoo-scripts sometimes create additional config-files.

Reply:
In that case, we should check if the commit-id declared is newer than the commit-id of the HEAD? The timestamp of a file is a local attribute and it corresponds to the time the file was checked out/cloned and it has nothing to do with the git timestamp of that file or HEAD.

Reply:
Updated